### ISSUE TITLE
Cannot download images: Error response from daemon: unauthorized: authentication required.

### ISSUE THREAD

**Issue by mxro** [docker/hub-feedback#1692]
## Problem description

Upon pulling an image from Docker Hub which exists an error as the following is reported:

```[stderr]docker: Error response from daemon: unauthorized: authentication required.
```

This error occurs intermittently. Eg sometimes downloading the image succeeds and sometimes it fails. 

This occurred for the following images:

```nginx:1.15.6
nginx:1.13.12
phpmyadmin/phpmyadmin:4.8.0
php:7.2.11-apache
```

Docker client used:

```Docker version 18.06.1-ce, build e68fc7a215d7133c34aa18e3b72b4a21fd0c6136```

Here some error messages received:

```[stderr]Unable to find image 'nginx:1.15.6' locally
[stderr]docker: Error response from daemon: unauthorized: authentication required.```

```[stderr]Unable to find image 'nginx:1.13.12' locally
[stderr]docker: Error response from daemon: unauthorized: authentication required.
[stderr]See 'docker run --help'.```

```[stderr]Unable to find image 'phpmyadmin/phpmyadmin:4.8.0' locally
[stderr]docker: Error response from daemon: unauthorized: authentication required.
[stderr]See 'docker run --help'.
```

```[stdout]Step 1/26 : FROM php:7.2.11-apache
[stderr]error parsing HTTP 404 response body: invalid character 'p' after top-level value: "404 page not found\n"```

**Comment by manishtomar** [comment]
Possible duplicate of #1693 which has been fixed. Please let us know if this occurs again.

**Comment by Gladskih** [comment]
It's not fixed in Docker version 19.03.1, build 74b1e89

**Comment by michealjroberts** [comment]
This issue also happens when trying to `docker login` to an Azure container registry, even tho the username and password supplied are correct.
